### ISSUE TITLE
Suggest to add document preview 

### ISSUE THREAD

**Issue by remi912** [danmunn/redmine_dmsf#1203]
Hi

I'm currently using Redmine with DMSF plugin, and I don't find a plugin which allows me to preview DMSF document.
I imagine that it could be a very nice feature to this document management system.
So I’d be very grateful if you could add this idea to your new patch.

Thank you in advance

**Comment by picman** [comment]
Please add in a few points how the preview should look like.

**Comment by remi912** [comment]
I think, it should look like to that : 

![image](https://user-images.githubusercontent.com/35763941/105712753-8af48080-5f1a-11eb-902f-ccc0718fb874.png)

Or that (if it's a picture) : 

![image](https://user-images.githubusercontent.com/35763941/105713239-2dacff00-5f1b-11eb-88d9-f431e156256d.png)

What do you think about that ?

**Comment by picman** [comment]
What is the purpose of this feature? You can simply click on the document and the document will be opened in the browser or in the associated application. Why should the app duplicate the existing functionality of other apps?

**Comment by remi912** [comment]
Yes, but only for PDF and pictures. The purpose is to support other format such as .docx, xls, pptx, odt, ods, odp, ...  which are not currently supported, and which always open a new popup for download.
I'm sure that it's really important that a document management system like DMSF provide this kind of functionality.

**Comment by picman** [comment]
Hm, it won't be easy.

**Comment by jonathan-0** [comment]
Just a side note on this topic as I would also find it helpful, but possibly could leverage the existing [redmine_more_previews](https://github.com/HugoHasenbein/redmine_more_previews) plugin.  

While not surprising, with redmine_more_previews, non-PDFs/pictures will download if using DMSF, but the preview renders as expected for .docx, .pptx, etc., if using core Files functionality instead.

Regardless, thanks!

**Comment by ChrisABailey** [comment]
This is probably the same level of complexity but is there some way that if I put a lint to a DMSF file in an issue or a wiki page that it could interact with Redmine_more_previews the same way an attachment does?  Specifically if I run more previews and attach a word file and reference it as attachment:xxx it will appear as a link which when clicked will open as a preview in a new page but if I add a link to a dims file using the dams markup syntax and click it, it always downloads the file (unless it is a PDF or image).

I would like to make some key project documents available as quick links without having to download before viewing them.

**Comment by picman** [comment]
Implemented in #1353.
